File: src/main/kotlin/com/ks/culinario/data/entity/TokenEntity.kt
package com.ks.culinario.data.entity

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "tokens")
data class TokenEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false, unique = true)
    val token: String = "",

    @Enumerated(EnumType.STRING)
    val tokenType: TokenType = TokenType.ACCESS,

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    val user: UserEntity? = null,

    @Column(nullable = false)
    val expiryDate: Instant = Instant.now(),
    
    val revoked: Boolean = false
)

File: dotnet/Culinario.Infrastructure/Persistence/Entities/TokenEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Culinario.Infrastructure.Persistence.Entities;

public enum TokenType
{
    ACCESS,
    REFRESH
}

[Table("Tokens")]
public class TokenEntity
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long? Id { get; set; }

    [Required]
    public string Token { get; set; } = string.Empty;

    [Required]
    public TokenType TokenType { get; set; }

    public long? UserId { get; set; }

    [ForeignKey("UserId")]
    public UserEntity? User { get; set; }

    [Required]
    public DateTime ExpiryDate { get; set; }

    public bool Revoked { get; set; }
}
